internal/perf/perf.go

```diff
@@ -162,25 +176,29 @@ func (pr *perfRunner) eventLoop() (err error) {
 			var event fftypes.EventDelivery
 			json.Unmarshal(msgBytes, &event)
 
-			var workerId int
+			var workerID int
 
 			switch event.Type {
 			case fftypes.EventTypeBlockchainEventReceived:
-				workerId, err = pr.GetBlockchainEventValue(pr.cfg.Node, event.Reference.String())
+				if event.BlockchainEvent == nil {
```

Comment:
If `event.BlockchainEvent == nil`, the worker should die and not be able to proceed